samcli/local/docker/container.py

```diff
@@ -39,7 +39,7 @@
 
 LOG = logging.getLogger(__name__)
 
-CONTAINER_CONNECTION_TIMEOUT = float(os.environ.get("SAM_CLI_CONTAINER_CONNECTION_TIMEOUT", 20))
+CONTAINER_CONNECTION_TIMEOUT = float(os.environ.get("SAM_CLI_CONTAINER_CONNECTION_TIMEOUT", "20"))
```

Comment:
(Just as a small rant incoming, but this code is okay how you did it)

Conceptually this is weird, but I imagine this is the easiest way to do it.

Because we're passing a `"20"` (string) that will be converted to a number with `float(..)`.. But in an ideal world we just want to pass the number right away instead of creating the intermediate string. 

What we would want is to do something like

```
CONTAINER_CONNECTION_TIMEOUT = float(os.environ.get("SAM_CLI_CONTAINER_CONNECTION_TIMEOUT")) if "SAM_CLI_CONTAINER_CONNECTION_TIMEOUT" in os.environ else 20
```
(check if the envvar exists, and only in that case convert to float, otherwise pass the number directly)

but that's terrible code too, so we can just keep it with what you did here.